src/btree/bt_sync.c

```diff
@@ -553,8 +553,11 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
             if (walk == NULL)
                 break;
 
-            /* Traverse through the internal page for obsolete child pages. */
-            if (F_ISSET(walk, WT_REF_FLAG_INTERNAL)) {
+            /*
+             * Perform checkpoint cleanup when not in recovery phase by traverse through the
+             * internal page for obsolete child pages.
+             */
+            if (!F_ISSET(conn, WT_CONN_RECOVERING) && F_ISSET(walk, WT_REF_FLAG_INTERNAL)) {
```

Comment:
Do you think we can also skip when doing a checkpoint during connection close?

Reply:
Yes, we can avoid checkpoint cleanup during startup and shutdown flows both. I will do the change.

Reply:
Fixed in 986a65d along with preventing cleanup, added a condition to the skip function to prevent reading pages into the cache during startup and shutdown.